packages/cadl-ranch-specs/http/payload/xml/main.tsp

```diff
@@ -0,0 +1,195 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Sends and receives bodies in XML format.")
+@scenarioService("/payload/xml")
+namespace Payload.Xml;
+
+@doc("Contains fields of primitive types.")
+model SimpleModel {
+  name: string;
+  age: int32;
+}
+
+@doc("Contains fields of arrays of primitive types.")
+model ModelWithSimpleArrays {
+  colors: string[];
+  counts: int32[];
+}
+
+@doc("Contains an array of models.")
+model ModelWithArrayOfModel {
+  items: SimpleModel[];
+}
+
+@doc("Contains an optional field.")
+model ModelWithOptionalField {
+  item: string;
+  counts?: int32;
+}
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Returns an instance of SimpleModel.
+Expected response body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+```
+""")
+@get
+op getSimpleModel(): {
+  @header("content-type") contentType: "application/xml";
+  @body body: SimpleModel;
+};
+
+@scenario
+@route("/simpleModel")
+@scenarioDoc("""
+Sends an instance of SimpleModel.
+Expected request body:
+```xml
+<SimpleModel>
+  <Name>foo</Name>
+  <Age>123</Age>
+</SimpleModel>
+```
+""")
+@put
+op putSimpleModel(
+  @header("content-type") contentType: "application/xml",
+  @body input: SimpleModel,
+): void;
+
+@scenario
+@route("/modelWithSimpleArrays")
+@scenarioDoc("""
+Returns an instance of ModelWithSimpleArrays.
+Expected response body:
+```xml
+<ModelWithSimpleArrays>
+  <Colors>red</Colors>
+  <Colors>green</Colors>
+  <Colors>blue</Colors>
+  <Counts>1</Counts>
+  <Counts>2</Counts>
+</ModelWithSimpleArrays>
+```
+""")
+@get
+op getModelWithSimpleArrays(): {
+  @header("content-type") contentType: "application/xml";
+  @body body: ModelWithSimpleArrays;
+};
+
+@scenario
+@route("/modelWithSimpleArrays")
+@scenarioDoc("""
+Sends an instance of ModelWithSimpleArrays.
+Expected request body:
+```xml
+<ModelWithSimpleArrays>
+  <Colors>red</Colors>
+  <Colors>green</Colors>
+  <Colors>blue</Colors>
+  <Counts>1</Counts>
+  <Counts>2</Counts>
+</ModelWithSimpleArrays>
+```
+""")
+@put
+op putModelWithSimpleArrays(
+  @header("content-type") contentType: "application/xml",
+  @body input: ModelWithSimpleArrays,
+): void;
+
+@scenario
+@route("/modelWithArrayOfModel")
+@scenarioDoc("""
+Returns an instance of ModelWithArrayOfModel.
+Expected response body:
+```xml
+<ModelWithArrayOfModel>
+  <SimpleModel>
```

Comment:
same here should always wrap, doc for example https://typespec.io/docs/next/libraries/xml/guide

Reply:
fixed